frontend/javascripts/dashboard/advanced_dataset/create_explorative_modal.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import { Modal, Radio, Button, Checkbox, Tooltip, Slider, Spin } from "antd";
+import { InfoCircleOutlined } from "@ant-design/icons";
+import React, { useState } from "react";
+import type { APIDatasetId } from "types/api_flow_types";
+import { Link } from "react-router-dom";
+import { useFetch } from "libs/react_helpers";
+import { getDataset } from "admin/admin_rest_api";
+
+import {
+  getSegmentationLayer,
+  doesSupportVolumeWithFallback,
+  getDatasetResolutionInfo,
+} from "oxalis/model/accessors/dataset_accessor";
+
+type Props = {
+  datasetId: APIDatasetId,
+  onClose: () => void,
+};
+
+const CreateExplorativeModal = ({ datasetId, onClose }: Props) => {
+  const dataset = useFetch(() => getDataset(datasetId), null, [datasetId]);
+  const [annotationType, setAnnotationType] = useState("hybrid");
+  const [userDefinedWithFallback, setUserDefinedWithFallback] = useState(true);
+  const [userDefinedResolutionIndices, setUserDefinedResolutionIndices] = useState([0, 10000]);
+
+  let modalContent = <Spin />;
+
+  if (dataset !== null) {
+    const segmentationLayer = getSegmentationLayer(dataset);
+
+    const datasetResolutionInfo = getDatasetResolutionInfo(dataset);
```

Comment:
I think, this needs to be specific to the segmentation layer's resolutions (if one exists). For example, consider this dataset:

```
color mags: 1, 2, 4, 8
segmentation mags: 4, 8
```

the user should only be able to select a range from mag 4 to 8. however, `getDatasetResolutionInfo` will return 1 to 8.

Reply:
The trouble here is that this depends on what the user has selected above. What you say is true in case of a fallback layer. I think it may be confusing to change the available range depending on the selections above. This may change the _selected_ range, which yields the question if that should be undone when the above selection is undone.
With the current code, the back-end will still select only those mags that are present in the fallback layer (intersection of available mags and selected range).
This is one of the check-boxes I added as “follow-up” in the PR description. If you think the current behavior is the more confusing one (mags that were selected can become unavailable in the end result), I’ll change it to the switching variant.

Reply:
> This is one of the check-boxes I added as “follow-up” in the PR description. 

Ah, I missed that :+1: Thanks for explaining the rationale of the current approach.

> I think it may be confusing to change the available range depending on the selections above. This may change the selected range, which yields the question if that should be undone when the above selection is undone.

I understand the problem, but think that the actual negative impact is negligible. The happy (and imo also likely) workflow will be that the user decides on fallback/no-fallback first and afterwards (if at all) changes the resolution restrictions. If they go back and fiddle with the fallback property again, the automated change of the resolutions can be annoying, but at least the UI reflects that and thus it's fixable.
The downside of the current approach is that the user selects something in the modal, hits "create" and then gets an annotation which does not reflect the settings which were specified. I think, this mismatch can be very confusing and annoying (probably, the user will think they made a mistake and retry). I think, it would be better to avoid this.

> which yields the question if that should be undone when the above selection is undone.

I don't think, that this is necessary.

Btw, the other "potential follow-ups" don't need to be in this PR imo, but the above thing would be good to have here.

Do you agree? Happy to change my mind about this :)

Reply:
Sounds fair :+1: